src/modules/QtQuick/Font.js

```diff
@@ -7,24 +7,50 @@ registerQmlType({
   constructor(parent) {
     super(parent);
     createProperty("bool", this, "bold");
-    createProperty("enum", this, "capitalization");
+    createProperty("enum", this, "capitalization", { initialValue: Font.MixedCase });
     createProperty("string", this, "family", { initialValue: "sans-serif" });
     createProperty("bool", this, "italic");
     createProperty("real", this, "letterSpacing");
     createProperty("int", this, "pixelSize", { initialValue: 13 });
     createProperty("real", this, "pointSize", { initialValue: 10 });
     createProperty("bool", this, "strikeout");
     createProperty("bool", this, "underline");
-    createProperty("enum", this, "weight");
+    createProperty("enum", this, "weight", { initialValue: Font.Normal });
     createProperty("real", this, "wordSpacing");
 
     this.$sizeLock = false;
 
+    const weightToCss = (weight) => {
+      switch (weight) {
+        case Font.Thin:
+          return "100";
+        case Font.ExtraLight:
+          return "200";
+        case Font.Light:
+          return "300";
+        case Font.Normal:
+          return "400";
+        case Font.Medium:
+          return "500";
+        case Font.DemiBold:
+          return "600";
+        case Font.Bold:
+          return "700";
+        case Font.ExtraBold:
+          return "800";
+        case Font.Black:
+          return "900";
+        default:
+          return "normal";
+      }
+    }
+
     this.boldChanged.connect(this, this.$onBoldChanged);
         this.capitalizationChanged.connect(function(newVal) {
+            newVal = ["none", "uppercase", "lowercase", "small-caps", "capitalize"][newVal]
```

Comment:
I think I've now made this function clearer. It was confusing before will all of the random conditional overwrites/uses of newVal.